ocaml/xapi/repository.ml

```diff
@@ -750,8 +750,12 @@ let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
   (* Evaluate recommended/pending guidances *)
   let recommended_guidances, pending_guidances =
     let recommended_guidances' =
-      eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind:Recommended
-        ~livepatches:successful_livepatches ~failed_livepatches
+      (* EvacuateHost will be applied before applying updates *)
+      List.filter
+        (fun g -> not (g = Guidance.EvacuateHost))
```

Comment:
Use `fun guidance -> guidance <> Guidance.EvacuateHost`; can be also written as `((<>) Guidance.EvacuateHost)` but it's a bit cryptic.

Does this make `EvacuateHost` obsolete? Can this guidance still reach a customer? The code also suggests to me it would be better to use sets than lists to compute these (below).

Reply:
> Does this make EvacuateHost obsolete? Can this guidance still reach a customer?
`EvacuateHost` guidance is available to customer before applying update with the HTTP endpoint "/updates", and the client (XenCenter) should handle the guidance`EvacuateHost` before applying update.

> The code also suggests to me it would be better to use sets than lists to compute these (below).
Yes, but I would like not to change this in this PR. :)